staging/src/k8s.io/apiserver/pkg/server/options/tracing.go

```diff
@@ -65,20 +79,24 @@ func (o *TracingOptions) ApplyTo(es *egressselector.EgressSelector, c *server.Co
 		return nil
 	}
 
```

Comment:
@dashpole - this is pre-existing, but I'm wondering if this is the point at which we should be catching an attempt to enable tracing in the API server if the feature is not enabled

That would mean if o.ConfigFile != "", and we got here, and the feature wasn't enabled, we'd return an error.

That would also mean calls to `TracingOptions#ApplyTo` should not be feature-gated

Reply:
I haven't seen that pattern (validating that feature-gates are enabled if the corresponding config is enabled), but it makes sense to me.

Reply:
otherwise, they can set a flag or config field, the binary silently ignores it, and then their identical invocation breaks on upgrade then feature leaps into action

Reply:
not sure you wanted this change in this PR, but added it in the last commit
PTAL, thanks